tests/nativeapp/test_version_create_processor.py

```diff
@@ -228,6 +229,63 @@ def test_process_no_version_from_user_no_version_in_manifest(
     mock_version_info_in_manifest.assert_called_once()
 
 
+# Test version create when user passed in a version and patch AND version does not exist in app package
+@mock.patch(
+    f"{VERSION_MODULE}.{CREATE_PROCESSOR}.get_existing_version_info", return_value=None
+)
+@pytest.mark.parametrize(
+    "policy_param", [allow_always_policy, ask_always_policy, deny_always_policy]
+)
+def test_process_no_version_exists_throws_bad_option_exception_one(
+    mock_existing_version_info, policy_param, temp_dir
+):
+    current_working_directory = os.getcwd()
+    create_named_file(
+        file_name="snowflake.yml",
+        dir=current_working_directory,
+        contents=[mock_snowflake_yml_file],
+    )
+
+    processor = _get_version_create_processor()
+    with pytest.raises(BadOptionUsage):
+        processor.process(
+            version="v1",
+            patch="12",
+            policy=policy_param,
+            git_policy=policy_param,
+            is_interactive=False,
+        )  # last three parameters do not matter here, so it should succeed for all policies.
+
+
+# Test version create when user passed in a version and patch AND app package does not exist
+@mock.patch(
+    f"{VERSION_MODULE}.{CREATE_PROCESSOR}.get_existing_version_info",
+    side_effect=ApplicationPackageDoesNotExistError("app_pkg"),
+)
+@pytest.mark.parametrize(
+    "policy_param", [allow_always_policy, ask_always_policy, deny_always_policy]
+)
+def test_process_no_version_exists_throws_bad_option_exception_two(
```

Comment:
ditto, what's _one and _two here?

Reply:
Replying for similar comments in one:
When you call `get_existing_version_info`, `BadOptionUsage` can be thrown in two situations: no existing version, or no existing app package. Hence the `one` and `two` to keep test names relatively short. But that is why each test has a comment above (like 260) describing what it tests.